pkg/sql/logictest/logic.go

```diff
@@ -3249,6 +3249,13 @@ func (t *logicTest) processSubtest(
 				}
 			}
 			cleanupUserFunc := t.setUser(fields[1], nodeIdx)
+			// In multi-tenant tests, we may need to also create database test when
+			// we switch to a different tenant.
+			if t.cfg.useTenant && strings.HasPrefix(fields[1], "host-cluster-") {
+				if _, err := t.db.Exec("CREATE DATABASE IF NOT EXISTS test; USE test;"); err != nil {
+					t.Fatal(err)
```

Comment:
Since we return an error from this function, do we want to `return errors.Wrapf` instead here?